src/script/interpreter.cpp

```diff
@@ -2137,11 +2166,49 @@ bool GenericTransactionSignatureChecker<T>::CheckUnvaultTriggerOutputsCommon(
     if (!this->txdata) return HandleMissingData(m_mdb);
     const auto& txd{*this->txdata};
 
-    if (!CommonVaultSpendChecks(txd, this->txTo->vout)) {
+    const auto num_outs = this->txTo->vout.size();
+    if (num_outs < 1 || num_outs > 3) {
         return false;
     }
 
-    unvault_output_spk = this->txTo->vout[0].scriptPubKey;
+    CAmount total_val_out_to_vaults{0};
+
+    assert(txd.m_spent_outputs_ready);
+    const CScript& vault_spk = txd.m_spent_outputs[this->nIn].scriptPubKey;
+
+    bool has_ea{false};
+    bool has_revault{false};
+
+    // Verify that all optional outputs are either an ephemeral anchor or a revault.
+    // Permit the optional outputs to appear in any order.
```

Comment:
In c00dad503d7dfa990afd24c9fde2555482cc247b "consensus: allow one revault output during unvault"

I don't see why the OP_UNVAULT output has to be the 0th output if the optional ones can be in any order.

Reply:
Fixed. OP_UNVAULT outputs (and recovery outputs) are now specified explicitly by an index on the witness stack. Thanks for this feedback.